linkerd/protocol/http/src/test/scala/io/buoyant/linkerd/protocol/http/ZipkinTracePropagatorTest.scala

```diff
@@ -0,0 +1,305 @@
+package io.buoyant.linkerd.protocol.http
+
+import io.buoyant.linkerd.protocol.ZipkinTrace
+import io.buoyant.linkerd.protocol.ZipkinTracePropagator
+
+import com.twitter.finagle.http.Request
+
+import org.scalatest.FunSuite
+
+class ZipkinTracePropagatorTest extends FunSuite {
+
+  test("get traceid from a b3 single header - empty header") {
+    // b3:
+    val ztp = new ZipkinTracePropagator()
+    val req = Request()
+    req.headerMap.add("b3", "")
+
+    val trace = ztp.traceId(req)
+    // this should not have returned a traceId because there's no span
+    assert(trace.isEmpty)
+    //b3 has been removed
+    assert(req.headerMap.get("b3").isEmpty)
+  }
+
+  test("get traceid from a b3 single header - one field - don't sample - b3: 0") {
+    //b3: 0
+    val ztp = new ZipkinTracePropagator()
+    val req = Request()
+    req.headerMap.add("b3", "0")
+
+    val trace = ztp.traceId(req)
+
+    // this should not have returned a traceId because there's no span
+    assert(trace.isEmpty)
+
+    //b3 has been removed
+    assert(req.headerMap.get("b3").isEmpty)
+    // check "x-b3-sampled" has been added to request
+    assert(req.headerMap.keys == Set("x-b3-sampled"))
+
+    // expect to get the right sampled value which is 0
+    var sampler = ZipkinTrace.getSampler(req.headerMap)
+    sampler match {
```

Comment:
Instead of doing explicit `assert(false)` whenever you have an empty option, I think you can simply combine the two with the contains combinator such as `assert(sampler.contains(0.0f))`. This will be a lot more concise. That stands true for all the other places, were you yous this sort of pattern

Reply:
Done, used assert(sampler.contains(0.0f)), much cleaner, thanks